File: Data Structures & Algorithms/remove-element/submission-5.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int r = nums.length - 1;
        int l = 0;
        int temp;
        for (int i=0; i<nums.length; i++){
            if (nums[i] == val){
                nums[i] = -1;
            } 
        }
        while (l <= r) {
            if (nums[l] < 0 ){
                if (nums[r] >= 0) {
                    temp = nums[l];
                    nums[l]=nums[r];
                    nums[r]=temp;
                    l++;
                    r--;
                } else {
                    r--;
                }              
            } else {
                l++;
            }           
        }
        return r+1;
}
}

File: Data Structures & Algorithms/sort-an-array/submission-5.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        self.quick_sort(nums, 0, len(nums) - 1)
        return nums

    def quick_sort(self, nums, lo, hi):
        if lo >= hi:
            return

        p = self.partition(nums, lo, hi)

        self.quick_sort(nums, lo, p)
        self.quick_sort(nums, p + 1, hi)

    def partition(self, nums, lo, hi):
        pivot = nums[(lo + hi) // 2]

        i = lo
        j = hi

        while True:

            while nums[i] < pivot:
                i += 1

            while nums[j] > pivot:
                j -= 1

            if i >= j:
                return j

            nums[i], nums[j] = nums[j], nums[i]

            i += 1
            j -= 1
            
                



        